hail/python/hail/experimental/import_gtf.py

```diff
@@ -160,3 +162,86 @@ def import_gtf(path, reference_genome=None, skip_invalid_contigs=False, min_part
     ht = ht.key_by('interval')
 
     return ht
+
+
+@typecheck(gene_symbols=nullable(sequenceof(str)),
+           gene_ids=nullable(sequenceof(str)),
+           transcript_ids=nullable(sequenceof(str)),
+           verbose=bool, reference_genome=nullable(reference_genome_type), gtf_file=nullable(str))
+def get_gene_intervals(gene_symbols=None, gene_ids=None, transcript_ids=None,
+                       verbose=True, reference_genome=None, gtf_file=None):
+    """Get intervals of genes or transcripts.
+
+       Get the boundaries of genes or transcripts from a GTF file, for quick filtering of a Table or MatrixTable.
+
+       Gencode v19 (GRCh37) GTF available at: gs://hail-common/references/gencode/gencode.v19.annotation.gtf.bgz
+       Gencode v29 (GRCh38) GTF available at: gs://hail-common/references/gencode/gencode.v29.annotation.gtf.bgz
+
+       Example
+       -------
+       >>> hl.filter_intervals(ht, get_gene_intervals(gene_symbols=['PCSK9']))  # doctest: +SKIP
+
+       Parameters
+       ----------
+
+       gene_symbols : :obj:`list` of :obj:`str`, optional
+           Gene symbols (e.g. PCSK9).
+       gene_ids : :obj:`list` of :obj:`str`, optional
+           Gene IDs (e.g. ENSG00000223972).
+       transcript_ids : :obj:`list` of :obj:`str`, optional
+           Transcript IDs (e.g. ENSG00000223972).
+       verbose : :obj:`bool`
+           If ``True``, print which genes and transcripts were matched in the GTF file.
+       reference_genome : :obj:`str` or :class:`.ReferenceGenome`, optional
+           Reference genome to use (passed along to import_gtf).
+       gtf_file : :obj:`str`
+           GTF file to load. If none is provided, but `reference_genome` is one of
+           `GRCh37` or `GRCh38`, a default will be used.
+
+       Returns
+       -------
+       :obj:`list` of :class:`.Interval`
+    """
+    GTFS = {
+        'GRCh37': 'gs://hail-common/references/gencode/gencode.v19.annotation.gtf.bgz',
+        'GRCh38': 'gs://hail-common/references/gencode/gencode.v29.annotation.gtf.bgz',
+    }
+    if reference_genome is None:
+        reference_genome = hl.default_reference().name
+    if gtf_file is None:
+        gtf_file = GTFS.get(reference_genome)
+        if gtf_file is None:
+            raise ValueError('get_gene_intervals requires a GTF file, or the reference genome be one of GRCh37 or GRCh38')
+    if not gene_symbols and not gene_ids and not transcript_ids:
```

Comment:
users who pass an empty list may be surprised at this error due to python truthiness

Reply:
? `[]` is not True, which is what you want here too, no? (if you pass 3 empty lists, this will return an empty list, which will then fail in `filter_intervals`)

Reply:
I'm not sure. I'd actually expect that if I pass 3 empty lists, I get an empty dataset back instead of an error. If I'm using this in a larger library where I'm always, say, passing `gene_symbols=compute_gene_symbols_from_function_input())`, then empty gene symbols seems like just the base case here, and it should do what it's asked (return no variants)

But up to you, this is your function

Reply:
good point